src/common/catalog-entities/kubernetes-cluster.ts

```diff
@@ -101,38 +100,32 @@ export class KubernetesCluster extends CatalogEntity<CatalogEntityMetadata, Kube
   }
 
   async onContextMenuOpen(context: CatalogEntityContextMenuContext) {
-    context.menuItems = [
+    context.menuItems.push(
       {
         title: "Settings",
         onlyVisibleForSource: "local",
-        onClick: async () => context.navigate(`/entity/${this.metadata.uid}/settings`)
+        onClick: () => context.navigate(`/entity/${this.metadata.uid}/settings`)
       },
-    ];
-
-    if (this.metadata.labels["file"]?.startsWith(ClusterStore.storedKubeConfigFolder)) {
```

Comment:
I think we need different confirm message for internally stored kubeconfigs vs user managed.

Reply:
How so? I don't delete the internally stored kubeconfigs. Just remove the `context`.

Reply:
I was thinking pretty much the opposite. When we modify user managed configuration we need to be super clear what we are doing. When we are modifying internal kubeconfig we can even remove whole file when a kubeconfig has no more contexts.

Reply:
I still don't understand why there needs to be a wording difference then. Unless you don't think that the current warning is good enough for the user-managed files.

I can add the logic for deleting the c&p files too.

Reply:
Ok, let's keep it as is. We can iterate this in follow-up PR (if needed).